homeassistant/components/unifiprotect/sensor.py

```diff
@@ -771,20 +771,22 @@ def __init__(
     def _async_update_device_from_protect(self, device: ProtectModelWithId) -> None:
         # do not call ProtectDeviceSensor method since we want event to get value here
         EventEntityMixin._async_update_device_from_protect(self, device)
-        if (
-            self.entity_description.ufp_smart_type
-            == SmartDetectObjectType.LICENSE_PLATE
-        ):
+        is_on = self.entity_description.get_is_on(device)
+        if is_on and self._event is not None:
             if (
-                self._event is None
-                or self._event.metadata is None
-                or self._event.metadata.license_plate is None
+                self.entity_description.ufp_smart_type
+                == SmartDetectObjectType.LICENSE_PLATE
             ):
-                self._attr_native_value = OBJECT_TYPE_NONE
-            else:
-                self._attr_native_value = self._event.metadata.license_plate.name
-        else:
-            if self._event is None:
-                self._attr_native_value = OBJECT_TYPE_NONE
+                if (
+                    self._event.metadata is not None
+                    and self._event.metadata.license_plate is not None
+                ):
+                    self._attr_native_value = self._event.metadata.license_plate.name
+                else:
+                    self._attr_native_value = OBJECT_TYPE_NONE
```

Comment:
This should actually not be possible to hit. If the event is a `smartDetectZone` with a License Plate detection, the metadata will not be none and there will be a license plate attached. There is just no easy way to model all of the possible subevent types with the metadata.